lib/matplotlib/colors.py

```diff
@@ -260,8 +267,35 @@ class ColorConverter(object):
         'k': (0.0, 0.0, 0.0),
         'w': (1.0, 1.0, 1.0), }
 
+    _prop_cycler = None
+
     cache = {}
 
+    @classmethod
+    def _get_nth_color(cls, val):
+        """
+        Get the Nth color in the current color cycle.  If N is greater
+        than the number of colors in the cycle, it is wrapped around.
+        """
+        from matplotlib.rcsetup import cycler
+        from matplotlib import rcParams
+
+        prop_cycler = rcParams['axes.prop_cycle']
+        if prop_cycler is None and 'axes.color_cycle' in rcParams:
+            clist = rcParams['axes.color_cycle']
+            prop_cycler = cycler('color', clist)
+
+        colors = prop_cycler.by_key()['color']
```

Comment:
As @tacaswell mentioned, this can be done in a backward compatible way (that doesn't require the latest version of cycler as this does).